### PR TITLE
DOC: Add a note about the ICD package when used in conda

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR adds one missing detail about the requirements for running on GPU.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

File: doc/sources/oneapi-gpu.rst [hunk 1]
```diff
@@ -67,6 +67,11 @@ Debian systems require installing package ``intel-opencl-icd`` (along with its d
 
 For Arch Linux, and for other distributions in general, see the `GPGPU article in the Arch wiki <https://wiki.archlinux.org/title/GPGPU>`__.
 
+.. important::
+    If using the |sklearnex| in a conda environment, GPU support requires the the OpenCL ICD package `for conda <https://github.com/IntelPython/intel-gpu-ocl-icd-system-feedstock>`__ to be installed in the conda environment, **in addition to the system install** of the same package: ::
```
Comment (icfaust) [inline doc/sources/oneapi-gpu.rst:71]: something is odd with the word "the" early on, if sklearnex is replaced scikit-learn extension then not a problem, but if its "sklearnex" then its funny.  I assume that if the user installs dpctl that this package would be included, (so this is an issue on standalone GPU support)

Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:71]: Here |sklearnex| expands to "Extension for Scikit-learn*", so it will say "the Extension for Scikit-learn*" when rendered.

> I assume that if the user installs dpctl that this package would be included

It is a transitive dependency for dpnp, but not for dpctl.

Comment (icfaust) [reply doc/sources/oneapi-gpu.rst:71]: thats a bit surprising.   Does that mean that the pip install of dpctl doesn't work out of the box with a SYCL gpu (I know there has been some back and forth with them in their issues page)?

Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:71]: If it's a pip install in a conda-managed environment, then that is indeed the case: it doesn't work out of the box.

**Review by Alexsandruss — APPROVED** [review]
